### PR TITLE
Add search and filters to infrastructure list (local data)

### PR DESCRIPTION
**Pull request by tremblaymath**
### Description of the Change

Add search on Name, Template, IPV6 and IPV4 
Add filter on  Name, Template, IPV6, IPV4, Status and Location
Add sorting to Name, Template and price

### Benefits

A standardized and simple way to do datagrid filtering

### CODE REVIEW

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js [hunk 1]
```diff
@@ -41,16 +60,38 @@ class CloudProjectComputeInfrastructureListCtrl {
             this.infra = infra;
             this.table.items = _.map(this.infra.vrack.publicCloud.items, instance => {
                 _.set(instance, "volumes", _.get(this.volumes, instance.id, []));
+                instance.ipv4 = instance.getPublicIpv4();
+                instance.ipv6 = instance.getPublicIpv6();
+                instance.macroRegion = this.RegionService.getMacroRegion(instance.region);
+                instance.statusToTranslate = this.getStatusToTranslate(instance);
+                instance.flavorTranslated = this.$translate.instant(`cpci_vm_flavor_category_${instance.flavor.name}`);
+
+                //patch for some translations that have &#160; html entities
```
Comment (tremblaymath) [inline client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: @Netnix Do you know why you have &#160; in flavor translations?

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.html [hunk 2]
```diff
@@ -63,23 +63,19 @@
...
+
     <section class="row cui-page__content">
-        <oui-datagrid data-rows="ComputeInfrastructureListCtrl.table.items" data-page-size="10">
-            <oui-column data-title="'cpci_vm_name' | translate" data-property="name" data-sortable>
-                <span data-ng-bind="$row.name"></span>
+        <oui-datagrid data-rows="ComputeInfrastructureListCtrl.table.items">
+            <oui-column title="'cpci_vm_name' | translate" data-property="name" data-type="string" data-searchable data-filterable data-sortable="asc">
             </oui-column>
-            <oui-column data-title="'cpci_vm_region' | translate" data-property="region" data-sortable>
+            <oui-column data-title="'cpci_vm_region' | translate" data-property="macroRegion" type="options" data-type-options="ComputeInfrastructureListCtrl.regionOptions" data-filterable>
```
Comment (tremblaymath) [inline client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.html]: filtering is on macroRegion but we display microRegion.

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.html [hunk 3]
```diff
@@ -97,24 +93,25 @@
...
             </oui-column>
-            <oui-column data-title="'cpci_vm_status' | translate" data-property="status">
+            <oui-column data-title="'cpci_vm_status' | translate" data-property="status" data-type="options" type-options="ComputeInfrastructureListCtrl.statusOptions" data-filterable>
                 <span class="label label-success"
-                      data-ng-if="$row.status === 'ACTIVE' && $row.monthlyBilling.status !== 'activationPending'">Ok</span>
+                      data-translate="cpci_vm_status_OK"
+                      data-ng-if="$row.statusToTranslate === 'OK'"></span>
                 <span class="label label-info"
                       data-translate="cpci_vm_status_UPDATING"
-                      data-ng-if="$row.status === 'ACTIVE' && $row.monthlyBilling.status === 'activationPending'"></span>
```
Comment (tremblaymath) [inline client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.html]: Logic moved to controller for centralization and possible reuse for sorting

File: client/components/cloud/region/region.service.js [hunk 4]
```diff
@@ -18,6 +18,21 @@ class RegionService {
         return number ? number[0] : "";
     }
 
+    getAllTranslatedMacroRegion () {
```
Comment (tremblaymath) [inline client/components/cloud/region/region.service.js]: Yes this list is hardcoded and should be updated when a new macro region is added.
We must add the corresponding translation before adding a new region, it cannot be generated dynamically.

File: client/app/cloud/project/compute/infrastructure/translations/Messages_fr_FR.xml [hunk 5]
```diff
@@ -56,6 +56,7 @@
    <translation id="cpci_vm_price" qtlid="83913">Prix</translation>
    <translation id="cpci_vm_hdd" qtlid="453731">Disque dur attaché</translation>
    <translation id="cpci_vm_status" qtlid="26443">Statut</translation>
+   <translation id="cpci_vm_status_OK">Ok</translation>
```
Comment (Netnix) [inline client/app/cloud/project/compute/infrastructure/translations/Messages_fr_FR.xml]: I prefer OK > Ok

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js [hunk 6]
```diff
@@ -41,16 +60,38 @@ class CloudProjectComputeInfrastructureListCtrl {
...
+                instance.flavorTranslated = instance.flavorTranslated .replace("&#160;"," ");
+
                 return instance;
             });
         }).catch(err => {
             this.table.items = [];
             this.CloudMessage.error(`${this.$translate.instant("cpci_errors_init_title")} : ${_.get(err, "data.message", "")}`);
+            this.$log.error(err);
```
Comment (Netnix) [inline client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: I'm not sure this is useful.

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js (hunk 1 above)
Comment (Netnix) [reply client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: I have no idea.

**Comment by Netnix** [comment]
This is a really cool feature.
Can you fix the issue where clicking on a action-menu will not show all items (because one of the items is hidden behind the navbar) ?
I wish we could search through location and status as well.

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js [hunk 7]
```diff
@@ -41,16 +60,38 @@ class CloudProjectComputeInfrastructureListCtrl {
...
         }).finally(() => {
             this.loaders.infra = false;
         });
     }
 
+    getStatusToTranslate(instance) {
```
Comment (neolitec) [inline client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: (optional) `switch/case` could be nice here.

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js (hunk 6 above)
Comment (tremblaymath) [reply client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: Yes we should log to output error somewhere for us else only the user see it in production. Sentry ideally. I will replace with $q.reject(), it's more appropriate

File: client/app/cloud/project/compute/infrastructure/translations/Messages_fr_FR.xml (hunk 5 above)
Comment (tremblaymath) [reply client/app/cloud/project/compute/infrastructure/translations/Messages_fr_FR.xml]: OK!

**Comment by tremblaymath** [comment]
Yes I will fix the action-menu in parallel(not blocking for this pr) and probably fix the height to 5.5 items.
Search for "enumeration" like status and location will probably be possible in the futur

File: client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js (hunk 6 above)
Comment (tremblaymath) [reply client/app/cloud/project/compute/infrastructure/list/cloud-project-compute-infrastructure-list.controller.js]: Sad... why errorOnUnhandledRejections is disabled?!
No logging, no sentry.. IN PRODUCTION
https://github.com/ovh-ux/ovh-manager-cloud/blob/master/client/components/q/q.config.js

**Comment by Netnix** [comment]
Done with this PR: https://github.com/ovh-ux/ovh-manager-cloud/pull/651

**Pull request closed without merging**